### PR TITLE
Homework

### PR DESCRIPTION
**Pull request by zoujing23**

### CODE REVIEW

**Comment by iphysresearch** [comment]
Good Job!!!   @zoujing23
---
Python homework:
- Total questions: 108
- Correct answers: 96
- Score: 88.00%
---
Numpy homework:
- Total questions: 10
- Correct answers: 10
- Score: 100.00%
---
Pandas homework:
- Total questions: 12
- Correct answers: 7
- Score: 58.00%
---

**Comment by iphysresearch** [comment]
Good Job!!!   @zoujing23
---
Python homework:
- Total questions: 108
- Correct answers: 108
- Score: 100.00%
---
Numpy homework:
- Total questions: 10
- Correct answers: 10
- Score: 100.00%
---
Pandas homework:
- Total questions: 12
- Correct answers: 7
- Score: 58.00%
---

**Comment by iphysresearch** [comment]
Good Job!!!   @zoujing23
---
Python homework:
- Total questions: 108
- Correct answers: 108
- Score: 100.00%
---
Numpy homework:
- Total questions: 10
- Correct answers: 10
- Score: 100.00%
---
Pandas homework:
- Total questions: 12
- Correct answers: 12
- Score: 100.00%
---

**Comment by iphysresearch** [comment]
Good Job!!!   @zoujing23
---
Python homework:
- Total questions: 108
- Correct answers: 108
- Score: 100.00%
---
Numpy homework:
- Total questions: 10
- Correct answers: 10
- Score: 100.00%
---
Pandas homework:
- Total questions: 12
- Correct answers: 12
- Score: 100.00%
---

**Comment by iphysresearch** [comment]
Good Job!!!   @zoujing23
---
Python homework:
- Total questions: 108
- Correct answers: 108
- Score: 100.00%
---
Numpy homework:
- Total questions: 10
- Correct answers: 10
- Score: 100.00%
---
Pandas homework:
- Total questions: 12
- Correct answers: 12
- Score: 100.00%
---

**Review by iphysresearch — CHANGES_REQUESTED** [review]
@zoujing23 

作业做得很好！建议把 `2023/homework_python.txt` 从你的PR中删掉吧，保持你的修改文件都在你的个人作业目录中。

**Comment by iphysresearch** [comment]
Good Job!!!   @zoujing23
---

**Comment by iphysresearch** [comment]
Good Job!!!   @zoujing23
---

**Comment by zoujing23** [comment]
#434
#将不是字母的字符换成空格
for x in s:
    if x.isalpha():
         continue
    else:
         s=s.replace(x,' ')
#分割字符串，将其储存为列表
l=s.split()
print(len(l))#输出字符的个数
print('The',len(l),'segments are',l)

#1869
#初始化a，b用于储存1，0的个数
a=0
b=0
#遍历字符串，分别找到1和0的个数
for i in [0,len(s)-1]:
    if s[i]=='1':
        a=a+1
    else:
        b=b+1
#对比a，b的大小后输出True or Flase
if a>b:
    print('True')
else:
    print('Flase')

#1784
#遍历字符串，找到1后若下一个仍是1则直接输出true，否则输出flase
for i in [0,len(s)-1]:
    if s[i]=='1' and s[i+1]=='1':
        print('true')
    else:
        print('flase')

#852
#l为第一个数字的位置，r为最后一个数字的位置
l, r = 0, len(A)-1
#直接用二分法取最中间的位置作为开始循环的起点
while l<r:
    mid = (l+r)//2
    if A[mid] > A[mid+1]:
        r= mid
#每一次失败都将第一个数字的位置往后挪一位，直到找到peak为止
    else:
        l= mid + 1
print(l)

#162
#peak即为数组中最大值
#遍历所有元素，打印出最大值所在的位置
for i in [0,len(A)-1]:
    if len[i]==max(A):
        print(i)
    else:
        break

File: 2023/homework/zoujing23/homework 12.08.ipynb
[the remark this answers is not in the document]
Comment (zoujing23) [reply 2023/homework/zoujing23/homework 12.08.ipynb:1]: 邹靓-国科大

**Comment by iphysresearch** [comment]
@zoujing23 
恭喜【邹靓-国科大】完成Python编程基础+扩展作业和全部的数据可视化作业！

**Pull request merged**